hack/update-codegen.sh

```diff
@@ -50,14 +54,28 @@ function generate_group() {
     --output-package ${INFORMERS_PKG} \
     --output-base ${SCRIPT_BASE}
 
+  # The following generators do not respect the --output-package, so generate
+  # their output into a temporary location and rsync it over.
+  local GEN_TMPDIR="$(mktemp -d /tmp/genXXXXXXXX)"
+
   echo "generating deep copies"
-  bazel run @io_k8s_code_generator//cmd/deepcopy-gen -- --input-dirs ${APIS_PKG}/${GROUP_NAME} --input-dirs ${APIS_PKG}/${GROUP_NAME}/${VERSION}
+  bazel run @io_k8s_code_generator//cmd/deepcopy-gen -- --input-dirs ${APIS_PKG}/${GROUP_NAME} --input-dirs ${APIS_PKG}/${GROUP_NAME}/${VERSION} --output-base ${GEN_TMPDIR}
 
   echo "generating defaults"
-  bazel run @io_k8s_code_generator//cmd/defaulter-gen -- --input-dirs ${APIS_PKG}/${GROUP_NAME} --input-dirs ${APIS_PKG}/${GROUP_NAME}/${VERSION}
+  bazel run @io_k8s_code_generator//cmd/defaulter-gen -- --input-dirs ${APIS_PKG}/${GROUP_NAME} --input-dirs ${APIS_PKG}/${GROUP_NAME}/${VERSION} --output-base ${GEN_TMPDIR}
 
   echo "generating conversions"
-  bazel run @io_k8s_code_generator//cmd/conversion-gen -- --input-dirs ${APIS_PKG}/${GROUP_NAME} --input-dirs ${APIS_PKG}/${GROUP_NAME}/${VERSION}
+  bazel run @io_k8s_code_generator//cmd/conversion-gen -- --input-dirs ${APIS_PKG}/${GROUP_NAME} --input-dirs ${APIS_PKG}/${GROUP_NAME}/${VERSION} --output-base ${GEN_TMPDIR}
+
+  rsync -a "${GEN_TMPDIR}/k8s.io/cluster-registry/" "${SCRIPT_BASE}/${SCRIPT_PACKAGE}"
+
+  if [[ "${GEN_TMPDIR}" == "/tmp/gen"* ]]; then
+    rm -rf "${GEN_TMPDIR}"
+  fi
 }
 
 generate_group clusterregistry v1alpha1
+
+if [[ "${TMP_GOPATH}" == "/tmp/gopath"* ]]; then
+  rm -rf "${TMP_GOPATH}"
+fi
```

Comment:
Do you want to make this a trap handler? See https://github.com/kubernetes/kubernetes/blob/04477ff90ceb55b864a4ac96fcc8c8b510414ead/staging/src/k8s.io/apiextensions-apiserver/hack/build-image.sh#L25 for an example.

Reply:
Good call. Done.